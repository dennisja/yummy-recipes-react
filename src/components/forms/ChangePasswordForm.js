import React, { Component } from 'react';
import User from '../../helpers/User';

export default class ChangePasswordForm extends Component {
    state = {
        current_password:"",
        new_password:"",
        new_password_again:"",
    }

    handleInputChange = (event)=>{
        const target = event.target;
        this.setState({
            [target.name]:target.value
        });
    }

    handleFormSubmit = (event) =>{
        event.preventDefault();
        alert('Yeah')
        //make api request to change password here
        User.changeUserPassword(this.state)
        .then(response=>{
            console.log(response);
        })
        .catch(error=>{
            console.log(error)
        })
    }

    render() {
        return (
          <form onSubmit={this.handleFormSubmit}>
            <div className="input-field">
              <label htmlFor="current_password">Current Password</label>
              <input type="password" name="current_password" id="current_password" onChange={this.handleInputChange} required />
            </div>
            <div className="input-field">
              <label htmlFor="new_password">New Password </label>
              <input type="password" name="new_password" id="new_password" onChange={this.handleInputChange} required />
            </div>
            <div className="input-field">
              <label htmlFor="new_password_again">Confirm New Password: </label>
              <input type="password" name="new_password_again" id="new_password_again" onChange={this.handleInputChange} required />
            </div>
            <div className="center-align">
                <input type="submit" className="btn btn-small orange" value="Change Password" name="change_password" id="change_pasword" />
            </div>
          </form>
        );
    }
}
