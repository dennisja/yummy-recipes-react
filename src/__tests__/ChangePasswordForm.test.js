import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import ChangePasswordForm from '../components/forms/ChangePasswordForm';

describe('Test ChangePasswordForm', () => {
    const formWrapper = shallow(<ChangePasswordForm />);
    const preventDefault = jest.fn();


    it('renders properly', () => {
        expect(shallowToJson(formWrapper)).toMatchSnapshot();
    });

    it('renders a form element', () => {
        expect(formWrapper.find('form')).toHaveLength(1);
    });

    it('has all form fields', () => {
        expect(formWrapper.find('#current_password')).toHaveLength(1);
        expect(formWrapper.find('#new_password')).toHaveLength(1);
        expect(formWrapper.find('#new_password_again')).toHaveLength(1);
        expect(formWrapper.find('#change_pasword')).toHaveLength(1);
    });

    it('submits data', () => {
        formWrapper.setState({ current_password: 'Hi', new_password: 'Hello', new_password_again: 'Hello' });
        formWrapper.find('form').simulate('submit', { preventDefault });
        expect(preventDefault).toBeCalled();
    });
});
