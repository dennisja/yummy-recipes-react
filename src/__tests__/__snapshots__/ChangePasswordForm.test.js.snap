// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ChangePasswordForm renders properly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="input-field"
  >
    <label
      htmlFor="current_password"
    >
      Current Password
    </label>
    <input
      id="current_password"
      name="current_password"
      onChange={[Function]}
      required={true}
      type="password"
    />
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="new_password"
    >
      New Password 
    </label>
    <input
      id="new_password"
      name="new_password"
      onChange={[Function]}
      required={true}
      type="password"
    />
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="new_password_again"
    >
      Confirm New Password: 
    </label>
    <input
      id="new_password_again"
      name="new_password_again"
      onChange={[Function]}
      required={true}
      type="password"
    />
  </div>
  <div
    className="center-align"
  >
    <input
      className="btn btn-small orange"
      id="change_pasword"
      name="change_password"
      type="submit"
      value="Change Password"
    />
  </div>
</form>
`;
